src/seedu/addressbook/ui/Formatter.java

```diff
@@ -0,0 +1,172 @@
+
+    /** Format of indexed list item */
+    private static final String MESSAGE_INDEXED_LIST_ITEM = "\t%1$d. %2$s";
+
+
+    /** Offset required to convert between 1-indexing and 0-indexing.  */
+    public static final int DISPLAYED_INDEX_OFFSET = 1;
+
+    /** Format of a comment input line. Comment lines are silently consumed when reading user input. */
+    private static final String COMMENT_LINE_FORMAT_REGEX = "#.*";
+
+    private final Scanner in;
+    private final PrintStream out;
+
+    public Formatter() {
+        this(System.in, System.out);
+    }
+
+    public Formatter(InputStream in, PrintStream out) {
+        this.in = new Scanner(in);
+        this.out = out;
+    }
+
+    /**
+     * Returns true if the user input line should be ignored.
+     * Input should be ignored if it is parsed as a comment, is only whitespace, or is empty.
+     *
+     * @param rawInputLine full raw user input line.
+     * @return true if the entire user input line should be ignored.
+     */
+    private boolean shouldIgnore(String rawInputLine) {
+        return rawInputLine.trim().isEmpty() || isCommentLine(rawInputLine);
+    }
+
+    /**
+     * Returns true if the user input line is a comment line.
+     *
+     * @param rawInputLine full raw user input line.
+     * @return true if input line is a comment.
+     */
+    private boolean isCommentLine(String rawInputLine) {
+        return rawInputLine.trim().matches(COMMENT_LINE_FORMAT_REGEX);
+    }
+
+    /**
+     * Prompts for the command and reads the text entered by the user.
+     * Ignores empty, pure whitespace, and comment lines.
+     * Echos the command back to the user.
+     * @return command (full line) entered by the user
+     */
+    public String getUserCommand() {
+        out.print(LINE_PREFIX + "Enter command: ");
+        String fullInputLine = in.nextLine();
+
+        // silently consume all ignored lines
+        while (shouldIgnore(fullInputLine)) {
+            fullInputLine = in.nextLine();
+        }
+
+        showToUser("[Command entered:" + fullInputLine + "]");
+        return fullInputLine;
+    }
+
+
+    public void showWelcomeMessage(String version, String storageFilePath) {
+        String storageFileInfo = String.format(MESSAGE_USING_STORAGE_FILE, storageFilePath);
+        showToUser(
+                DIVIDER,
+                DIVIDER,
+                MESSAGE_WELCOME,
+                version,
+                MESSAGE_PROGRAM_LAUNCH_ARGS_USAGE,
+                storageFileInfo,
+                DIVIDER);
+    }
+
+    public void showGoodbyeMessage() {
+        showToUser(MESSAGE_GOODBYE, DIVIDER, DIVIDER);
+    }
+
+
+    public void showInitFailedMessage() {
+        showToUser(MESSAGE_INIT_FAILED, DIVIDER, DIVIDER);
+    }
+
+    /** Shows message(s) to the user */
+    public void showToUser(String... message) {
+        for (String m : message) {
+            out.println(LINE_PREFIX + m.replace("\n", LS + LINE_PREFIX));
```

Comment:
Your formatter is still doing both formatting and printing. This defeats the purpose of segregating the 2 classes.